src/settings/eeprom/main_board.rs

```diff
@@ -148,65 +146,57 @@ pub struct SerializedMainBoardData {
     pub fan_speed: f32,
 }
 
-impl From<BoosterMainBoardData> for SerializedMainBoardData {
-    fn from(d: BoosterMainBoardData) -> Self {
-        Self {
-            version: d.version,
-            ip: d.ip,
-            broker: d
-                .broker
-                .parse()
-                .unwrap_or_else(|_| IpAddr::new(&[10, 0, 0, 2])),
-            gateway: d.gateway,
-            netmask: d.netmask,
-            id: MqttIdentifier(d.id),
-            fan_speed: d.fan_speed,
-        }
-    }
-}
-
 impl SerializedMainBoardData {
     fn with_mac(self, eui48: &[u8; 6]) -> BoosterMainBoardData {
         let mut broker = String::new();
         write!(&mut broker, "{}", self.broker.0).unwrap();
+        let netmask = smoltcp::wire::IpAddress::Ipv4(self.netmask.0);
+
+        let ipcidr = smoltcp::wire::IpCidr::new(
+            smoltcp::wire::IpAddress::Ipv4(self.ip.0),
+            netmask.prefix_len().unwrap_or(0),
+        );
+
+        let mut ip = String::new();
+        write!(&mut ip, "{}", ipcidr).unwrap();
```

Comment:
I'd ideally like to store the `IpCidr` directly in the `Settings` struct, but I don't know if miniconf currently allows for this with the `typ` flags since I would need to operate with owned `String<18>` types instead of refs